### ISSUE TITLE
It appears that can't use Argument instance when creating Field

### ISSUE THREAD

**Issue by davispuh** [rmosolgo/graphql-ruby#2459]
https://github.com/rmosolgo/graphql-ruby/blob/869484464958cd6c5ec286260d5d5f9ba4cb51b1/lib/graphql/schema/field.rb#L229

It appears that ^ line/case is not usable.

Because it's not possible to create `GraphQL::Schema::Argument` without specifying `owner`  which is required and isn't changeable later.

https://github.com/rmosolgo/graphql-ruby/blob/1c95c917837491e8350e94a18c3f9c4d3f22bc9b/lib/graphql/schema/argument.rb#L43

So basically, currently you can't do this
````ruby
args = {'arg' => GraphQL::Schema::Argument.new(required: false)} # needs owner but we haven't created field yet
GraphQL::Schema::Field.new(name: 'field', type: GraphQL::INT_TYPE, null: true, arguments: args)
````

**Comment by rmosolgo** [comment]
> currently you can't do this

That's right, I don't intend to support making a schema apart from the class-based DSL. Is there a good reason to do it that way instead of something like: 

```ruby 
args = { "arg" => { required: false } } 

MyType.field(:field, Integer, null: true) do 
  args.each do |name, config| 
    argument(name, config) 
  end 
end 
```

?

**Comment by davispuh** [comment]
Then probably should remove that line as it's unusable. I noticed it while looking how to dynamically create schema, fields and arguments.

Currently I'm doing it this way
```ruby


query = Class.new(GraphQL::Schema::Object)
query.graphql_name('Query')

params = { ... }
params[:arguments][argName] = { ... }

query.field(name, type, **params)

schema = Class.new(GraphQL::Schema)
schema.query(query)
````
